flixel/FlxObject.hx

```diff
@@ -1173,4 +1183,16 @@ class FlxObject extends FlxBasic
 	{
 		return allowCollisions = Value;
 	}
+	
+	private function set_path(path:FlxPath):FlxPath
+	{
+		if (this.path == path)
+			return path;
+		
+		if (this.path != null)
+			path.object = null;
```

Comment:
There's a bug here:
```if (this.path != null)
			path.object = null;
```
Should be:
```if (this.path != null)
			this.path.object = null;
```

With regards to destroy, I suggest we destroy the path that is being replaced (the one being nulled out).